specification/cosmos-db/resource-manager/readme.md

```diff
@@ -44,10 +44,16 @@ input-file:
 ``` yaml
 directive:
   - suppress: TrackedResourceGetOperation
+    from: cosmos-db.json
+    where: $.definitions.DatabaseAccountCreateUpdateParameters
     reason: The DatabaseAccount request and response resources differ so the DatabaseAccountCreateUpdateParameters must only have a PUT operation
   - suppress: TrackedResourcePatchOperation
+    from: cosmos-db.json
+    where: $.definitions.DatabaseAccountCreateUpdateParameters
     reason: The DatabaseAccount request and response resources differ so the DatabaseAccountCreateUpdateParameters must only have a PUT operation
   - suppress: DefinitionsPropertiesNamesCamelCase

```

Comment:
I talked with Gaurav and he is not sure why it is needed to also supress this rule.

Reply:
PATCH is not supported for all the properties in the DatabaseAccount resource today. We would have to create another resource for PATCH in particular until that gets deployed.

Reply:
He means the camel case rule

Reply:
There's a property within the Metrics API with a leading underscore which breaks the DefinitionsPropertiesNamesCamelCase rule

Reply:
@dmakwana - If patch is not supported for all properties, what is the issue in creating a new model which takes in only the properties supported by PATCH. I am not sure if I understand this. Is PATCH supported for a subset of properties on the service side?

Reply:
Yes PATCH is only supported for a subset of properties on the service side today. We plan to add the remaining properties at a future date. That is why we need separate resources for CreateUpdate (PUT), Patch (PATCH).

Reply:
Then why do you need a suppression? Suppression is only granted for things which are false positives due to tooling issue, or things which you will never fix due to certain reasons etc.

Reply:
The reason we need the suppression is because DatabaseAccountCreateUpdateParameters cannot have a PATCH method on it, since there are properties which are required. In lieu of this, we have created another resource DatabaseAccountPatchParameters which has all the properties as optional.

Reply:
Ok looks like a tooling issue @veronicagg to confirm if this can be fixed. And yes, if thats the case, granting suppression for this is fine.

Reply:
@ravbhatnagar @dmakwana is the issue that "DatabaseAccountCreateUpdateParameters" is not really a resource/tracked resource? seems that it's a model constructed to pass the parameters, but the actual ARM resource is "DatabaseAccount". If that's the case, the suppression reason should probably state this. 
@ravbhatnagar if you're ok with the api design being that way, then the suppression should be added. If the models should be shared or "DatabaseAccountCreateUpdateParameters" should not be doing an allOf on "Resource", then the violation may go away.

Reply:
@veronicagg The reason that is present already states why the actual tracked resource DatabaseAccount can not be used for the request and response. Since that's the case, a different model for GET/PUT/PATCH had to be created which implies that DatabaseAccountCreateUpdateParameters cannot support GET or PATCH, hence the two individual suppressions. DatabaseAccount -> GET  DatabaseAccountCreateUpdateParameters -> PUT and DatabaseAccountPatchParameters -> PATCH. I am aware of the API changes needed. It's in the roadmap